crates/bevy_ecs/src/entity/entity_set.rs

```diff
@@ -382,25 +382,24 @@ impl<I: Iterator<Item: TrustedEntityBorrow> + Debug> Debug for UniqueEntityIter<
 mod tests {
     use alloc::{vec, vec::Vec};
 
-    #[expect(unused_imports)]
     use crate::prelude::{Schedule, World};
 
-    #[expect(unused_imports)]
     use crate::component::Component;
     use crate::query::{QueryState, With};
     use crate::system::Query;
     use crate::world::Mut;
-    #[expect(unused_imports)]
     use crate::{self as bevy_ecs};
-    #[expect(unused_imports)]
-    use crate::{entity::Entity, world::unsafe_world_cell};
+    use crate::entity::Entity;
 
     use super::UniqueEntityIter;
 
     #[derive(Component, Clone)]
     pub struct Thing;
 
-    #[expect(clippy::iter_skip_zero)]
+    #[expect(
+        clippy::iter_skip_zero,
+        reason = "Uniqueness of entities should be preserved, even when skipping zero entities."
```

Comment:
I'm not actually convinced this is the reason... But I need to hear back from @Victoronz first, before removing the `.skip(0)`.